engine/src/proxy.rs

```diff
@@ -125,7 +127,7 @@ pub enum Snapshot {
     /// Memory snapshot
     Memory(MemorySnapshot),
```

Comment:
Shall we wrap `MemorySnapshot` into `metrics::Layer`?

Reply:
`MemorySnapshot` is currently only used in tests and is not necessary.